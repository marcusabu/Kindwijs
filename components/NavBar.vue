<template>
  <div>
    <b-navbar toggleable="lg" type="dark" variant="info">
      <b-navbar-brand href="/">Kindwijs home</b-navbar-brand>
      <b-navbar-toggle target="nav-collapse"></b-navbar-toggle>
      <b-collapse id="nav-collapse" is-nav>
        <b-navbar-nav>
          <b-nav-item href="/wanneer" :class="setActive('wanneer')">
            Wanneer een kindercoach
          </b-nav-item>
          <b-nav-item href="/werkwijze" :class="setActive('werkwijze')"
            >Werkwijze
          </b-nav-item>
          <b-nav-item href="/onderwijs" :class="setActive('onderwijs')"
            >Onderwijs
          </b-nav-item>
          <b-nav-item href="/over-mij" :class="setActive('over-mij')"
            >Over mij
          </b-nav-item>
          <b-nav-item href="/tarieven" :class="setActive('tarieven')"
            >Tarieven
          </b-nav-item>
          <b-nav-item href="/contact" :class="setActive('contact')"
            >Contact
          </b-nav-item>
        </b-navbar-nav>

        <!-- Right aligned nav items -->
        <b-navbar-nav class="ml-auto">
          <b-nav-item
            href="https://www.facebook.com/Kindwijs-Kindercoaching-101080354697170/"
            target="_blank"
            ><BIconFacebook
          /></b-nav-item>
          <b-nav-item
            href="https://www.linkedin.com/in/marie-louise-rosier-75a05042/"
            target="_blank"
            ><BIconLinkedin
          /></b-nav-item>
        </b-navbar-nav>
      </b-collapse>
    </b-navbar>
  </div>
</template>

<script>
import { BIconFacebook, BIconLinkedin } from 'bootstrap-vue'
export default {
  components: {
    BIconFacebook,
    BIconLinkedin,
  },
  methods: {
    setActive(routeName) {
      return routeName === this.$route.name ? 'active' : ''
    },
  },
}
</script>

<style scoped>
.navbar {
  background-color: #f4bb31 !important;
}

.nav-link {
  color: white !important;
}

.active {
  border-bottom: 3px solid white;
}
</style>
