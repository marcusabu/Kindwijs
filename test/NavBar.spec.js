import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import NavBar from '@/components/NavBar.vue'
import BootstrapVue from 'bootstrap-vue'

Vue.use(BootstrapVue)

const $route = {
  name: 'index',
}

describe('NavBar', () => {
  it('matches the snapshot', () => {
    const wrapper = shallowMount(NavBar, {
      mocks: {
        $route,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
