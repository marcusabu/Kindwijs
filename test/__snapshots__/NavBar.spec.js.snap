// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches the snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
